ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1144,6 +1138,7 @@ private void prepareForSending() {
      * Submits encrypt job for the given {@link PutChunk} and processes the callback for the same
      */
     private void encryptChunk() {
+      ByteBuf toEncrypt = buf.retain();
```

Comment:
It seems like this buffer is retained and then retained again with toEncrypt.retainedDuplicate() and then released at the end of this method. Could the retain from this line and release on 1204 be removed?

Reply:
legacy code.